climada/entity/exposures/litpop/litpop.py

```diff
@@ -0,0 +1,1199 @@
+            # loop over countries: litpop is initiated for each individual polygon
+            # within each country and combined at the end.
+            litpop_list = \
+                [self._set_one_country(country,
+                                       res_arcsec=res_arcsec,
+                                       exponents=exponents,
+                                       fin_mode=fin_mode,
+                                       total_value=total_values[idc],
+                                       reference_year=reference_year,
+                                       gpw_version=gpw_version,
+                                       data_dir=data_dir,
+                                       reproject_first=reproject_first)
+                 for idc, country in enumerate(countries)]
+        # make lists of countries with Exposure initaited and those ignored:
+        countries_in = \
+            [country for i, country in enumerate(countries) if litpop_list[i] is not None]
+        countries_out = \
+            [country for i, country in enumerate(countries) if litpop_list[i] is None]
+        if not countries_in:
+            raise ValueError('No valid country identified in %s, aborting.' % countries)
+        litpop_list = [exp for exp in litpop_list if exp is not None]
+        if countries_out:
+            LOGGER.warning('Some countries could not be identified and are ignored: ' +
+                           '%s. Litpop only initiated for: %s', countries_out, countries_in)
+
+        tag.description = ('LitPop Exposure for %s at %i as, year: %i, financial mode: %s, '
+                           'exp: [%i, %i], admin1_calc: %s'
+                           % (countries_in, res_arcsec, reference_year, fin_mode,
+                              exponents[0], exponents[1], str(admin1_calc)))
+
+        Exposures.__init__(
+            self,
+            data=Exposures.concat(litpop_list).gdf,
+            crs=litpop_list[0].crs,
+            ref_year=reference_year,
+            tag=tag,
+            value_unit=get_value_unit(fin_mode),
+            exponents = exponents,
+            gpw_version = gpw_version,
+            fin_mode = fin_mode,
+        )
+
+        try:
+            rows, cols, ras_trans = u_coord.pts_to_raster_meta(
+                (self.gdf.longitude.min(), self.gdf.latitude.min(),
+                 self.gdf.longitude.max(), self.gdf.latitude.max()),
+                u_coord.get_resolution(self.gdf.longitude, self.gdf.latitude))
+            self.meta = {
+                'width': cols,
+                'height': rows,
+                'crs': self.crs,
+                'transform': ras_trans,
+            }
+        except ValueError:
+            LOGGER.warning('Could not write attribute meta, because exposure'
+                           ' has only 1 data point')
+            self.meta = {'crs': self.crs}
+        self.check()
+
+    def set_nightlights(self, countries=None, shape=None, res_arcsec=15,
```

Comment:
There is a fin_mode `pop` in `set_country` because users should be able to explore the whole range of possible parameter combinations in `set_country` (while the wrapper `set_population` limits choices, for instance with regard to the choice iof `exponents`).
There is no fin_mode `nightlight` in `set_country`, because – other that for population – I see no value in redistributing nightlight intensity based on other exponent combinations.

So in essence, the difference is that population has two uses: it is both used for disaggregation and can be the target value (total value) to be disaaggregated if you want a population map as exposure. In contrast, nightlights are only used for disaggregation and not as total asset value. The method `set_nightlights` is for users who want to extract the raw nightlight intensity and do their own post-processing with it.

Does that make sense?

Reply:
Ok, thanks for explaining! Yes, I see why you have a `pop` option as finmode but no `nightlight`. But then you still provide a set_nightlight function to get this data loaded into the value columns of an exposures instance, even though you claim it wouldn't really be useful as an exposure by itself. 

So I'm just wondering, whether one really needs the set_nightlight altogether, then, or whether expert users who just want to tweak nightlight data cannot load a raster file into memory by themselves.

Reply:
Good point! I have added these two wrapper function due to expressed interest during the CLIMADA call, so it might be of interest to some users. Maybe I could make this special case more transparent by renaming `set_nightlights` to sth like `set_nightlight_intensity` ?

Reply:
How about get_nightlight_intensity?
But also happy as is ;)

Reply:
From my understanding, it is not a "set" but a "get" method (from a python perspective), that's the reason for the naming. I have added a warning in the method for users who could have the impression that the values represent asset values in USD instead of raw nightlight intensity.